File: ts-playwright-cucumber/src/pages/cartPage.ts
import type { Page } from 'playwright';
import { BasePage } from './basePage.ts';
import { CheckoutInformationPage } from './checkoutInformationPage.ts';

const CART_ITEM_NAMES = "[data-test='inventory-item-name']";
const CHECKOUT_BUTTON = "[data-test='checkout']";

export class CartPage extends BasePage {
  constructor(page: Page) {
    super(page);
  }

  listItemNames(): Promise<string[]> {
    return this.page.locator(CART_ITEM_NAMES).allTextContents();
  }

  async startCheckout(): Promise<CheckoutInformationPage> {
    await this.click(CHECKOUT_BUTTON);
    return new CheckoutInformationPage(this.page);
  }
}

File: java-selenium-junit5-cucumber/src/test/java/com/saucedemo/uitests/pages/CartPage.java
package com.saucedemo.uitests.pages;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public final class CartPage extends BasePage {

    private static final By CART_ITEM_NAMES = By.cssSelector("[data-test='inventory-item-name']");
    private static final By CHECKOUT_BUTTON = By.cssSelector("[data-test='checkout']");

    public CartPage(WebDriver driver) {
        super(driver);
    }

    public List<String> listItemNames() {
        return driver.findElements(CART_ITEM_NAMES).stream().map(WebElement::getText).toList();
    }

    public CheckoutInformationPage startCheckout() {
        click(CHECKOUT_BUTTON);
        return new CheckoutInformationPage(driver);
    }
}
